server/datastore/mysql/setup_experience.go

```diff
@@ -316,6 +320,43 @@ WHERE host_uuid = ?
 	return results, nil
 }
 
+func (ds *Datastore) UpdateSetupExperienceStatusResult(ctx context.Context, status *fleet.SetupExperienceStatusResult) error {
```

Comment:
nit: we'll want to add a test for this in setup_experience_test.go; can be done in a follow up tho as there are other tests to clean up in there.